### PR TITLE
reduce warnings in test_openclhpp.cpp file

### PR DESCRIPTION
**Pull request by pwisniewskimobica**
I done some fixes in test_openclhpp.cpp file 
due to reduce warnings

### CODE REVIEW

File: tests/test_openclhpp.cpp [hunk 1]
```diff
@@ -1561,7 +1561,7 @@ static cl_mem clCreateImage3D_testCreateImage3D_1_1(
     TEST_ASSERT_EQUAL(16, image_depth);
     TEST_ASSERT_EQUAL(256, image_row_pitch);
     TEST_ASSERT_EQUAL(65536, image_slice_pitch);
-    TEST_ASSERT_EQUAL_PTR((void *) 0xdeadbeef, host_ptr);
+    TEST_ASSERT_EQUAL_PTR(nullptr, host_ptr);
```
Comment (Mikolaj-Filar-Mobica) [inline tests/test_openclhpp.cpp:1564]: I don't think changing `0xdeadbeef` to `nullptr` is a good approach. I see the better solution is suggested in https://github.com/KhronosGroup/OpenCL-CLHPP/pull/223 by @Kerilk:
`    TEST_ASSERT_EQUAL_PTR((void *)(size_t)0xdeadbeef, host_ptr);
`
For this reason, this PR is also in conflict with the one mentioned above.

Comment (pwisniewskimobica) [reply tests/test_openclhpp.cpp]: Thanks 
You are right - changed back from nullptr to (void *)(size_t)0xdeadbeef
Other thing is that https://github.com/KhronosGroup/OpenCL-CLHPP/pull/223 
is not the same changes. This one and that are not conflicting- They are another fixes.

**Comment by bashbaug** [comment]
Is this still needed now that we merged https://github.com/KhronosGroup/OpenCL-CLHPP/pull/223?

**Comment by pwisniewskimobica** [comment]
@bashbaug 
I can say that https://github.com/KhronosGroup/OpenCL-CLHPP/pull/223 is another set of fixes.
For my it looks like it is designed mostly for linux while my fixes are for windows

**Comment by bashbaug** [comment]
I'm pretty sure Windows builds were also warning-free when we merged #223.  Here are the CI runs:

https://github.com/KhronosGroup/OpenCL-CLHPP/actions/runs/4799850369/jobs/8540025875

If we have additional warnings on Windows then they're either configurations we aren't testing (possible) or we've introduced new warnings recently (also possible).

**Comment by pwisniewskimobica** [comment]
@bashbaug basically you and me are right;)
I mean that you are right because CI does not shows warnings that I have on my local and I am right because Its strange but- the only difference I see is that CI uses cmake version 3.22.0 and I use 3.26
To proove I am attaching print screen
image

**Comment by Kerilk** [comment]
I think the confusion comes from this discrepancy between what is in the repository:
https://github.com/KhronosGroup/OpenCL-CLHPP/blob/0f420fb265d5d3a64d68deceabfd3689f7322e30/tests/test_openclhpp.cpp#L736
and what you have on your local machine (per your screenshot):
```
    for (int i = 0; i < num_devices; i++) {
```
So your rebase must only have been partial.

**Comment by pwisniewskimobica** [comment]
@bashbaug Thats right
That case I think this PR is not needed anymore
thanks

**Pull request closed without merging**